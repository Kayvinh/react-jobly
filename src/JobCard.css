.JobCard{
    margin-top: 1em;
}
