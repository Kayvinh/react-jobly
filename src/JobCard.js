import React, { useContext } from 'react';
import './JobCard.css';
import userContext from './userContext';

/** Renders information about a single job
 * 
 * Props
 * -job: {id, title, salary, equity, companyName}
 * -apply(): for applying to a job
 * 
 * State:
 * -none
 * 
 * JobCardList -> JobCard
 */
function JobCard({ id, title, companyName, salary, equity, apply }) {
  const { user } = useContext(userContext);
  const applications = user.applications;
  return (
    <div className='JobCard card'>
      <div className='card-body'>
        <div className='card-title'>
          <h6>
            <b>{title}</b>
            <p>{companyName}</p>
          </h6>
        </div>
        <small>
          <div>Salary: {salary}</div>
          <div>Equity: {equity}</div>
        </small>
        {applications.includes(id)
          ? <button className='btn btn-danger' disabled>Applied</button>
          : <button className='btn btn-danger' onClick={() => apply(id)}>Apply</button>
        }
      </div>
    </div>
  );
}

export default JobCard;