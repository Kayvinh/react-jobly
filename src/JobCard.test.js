import React from 'react';
import { render } from '@testing-library/react';
import JobCard from './JobCard';

const testJobCard = <JobCard title="test title"
    companyName="test company name"
    salary={100}
    equity={0.01} />;

test('renders without crashing', function () {
    render(testJobCard);
});

test('renders proper text', function () {
    const { container } = render(testJobCard);

    expect(container).toContainHTML("test title");
    expect(container).toContainHTML("test company name");
    expect(container).toContainHTML("Salary: 100");
    expect(container).toContainHTML("Equity: 0.01");
});

test('matches snapshot', function () {
    const { container } = render(testJobCard);

    expect(container).toMatchSnapshot()

});