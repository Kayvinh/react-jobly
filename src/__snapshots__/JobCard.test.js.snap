// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="JobCard card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title"
      >
        <h6>
          <b>
            test title
          </b>
          <p>
            test company name
          </p>
        </h6>
      </div>
      <small>
        <div>
          Salary: 
          100
        </div>
        <div>
          Equity: 
          0.01
        </div>
      </small>
    </div>
  </div>
</div>
`;
